tests/fiaas_deploy_daemon/crd/test_crd_watcher.py

```diff
@@ -229,6 +224,11 @@ def test_deploy(
             assert deployer_event == DeployerEvent(deployer_event_type, app_spec, None)
         assert deploy_queue.empty()
 
+    def test_delete(self, deploy_queue, watcher):
+        watcher.watch.return_value = [WatchEvent(DELETED_EVENT, FiaasApplication)]
+
```

Comment:
You'll have to call `crd_watcher._watch(None)` here, other than that it looks good.